Formula/redis@4.0.rb

```diff
@@ -0,0 +1,72 @@
+class RedisAT40 < Formula
+  desc "Persistent key-value database, with built-in net interface"
+  homepage "https://redis.io/"
+  url "http://download.redis.io/releases/redis-4.0.11.tar.gz"
+  sha256 "fc53e73ae7586bcdacb4b63875d1ff04f68c5474c1ddeda78f00e5ae2eed1bbb"
+
+  keg_only :versioned_formula
+
+  option "with-jemalloc", "Select jemalloc as memory allocator when building Redis"
+
+  def install
+    # Architecture isn't detected correctly on 32bit Snow Leopard without help
+    ENV["OBJARCH"] = "-arch #{MacOS.preferred_arch}"
+
+    args = %W[
+      PREFIX=#{prefix}
+      CC=#{ENV.cc}
+    ]
+    args << "MALLOC=jemalloc" if build.with? "jemalloc"
+    system "make", "install", *args
+
+    %w[run db/redis log].each { |p| (var/p).mkpath }
+
+    # Fix up default conf file to match our paths
+    inreplace "redis.conf" do |s|
+      s.gsub! "/var/run/redis.pid", var/"run/redis.pid"
+      s.gsub! "dir ./", "dir #{var}/db/redis/"
+      s.sub!  /^bind .*$/, "bind 127.0.0.1 ::1"
+    end
+
+    etc.install "redis.conf"
+    etc.install "sentinel.conf" => "redis-sentinel.conf"
+  end
+
+  plist_options :manual => "redis-server #{HOMEBREW_PREFIX}/etc/redis.conf"
```

Comment:
I believe this one needs to be updated to opt. See [cassandra@2.1](https://github.com/Homebrew/homebrew-core/blob/e0b944b3ccb73fa1aaea99dc94b57423dd278828/Formula/cassandra%402.1.rb#L97) for example.